community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/logical/greedy/join.scala

```diff
@@ -42,7 +42,7 @@ object join extends CandidateGenerator[GreedyPlanTable] {
         if (shared.nonEmpty && shared.forall(isApplicable)) {
 
           val hints = queryGraph.hints.collect {
-            case hint@UsingJoinHint(identifier) if shared contains IdName(identifier.name) => hint
+            case hint@UsingJoinHint(identifiers) if identifiers.forall(identifier => shared contains IdName(identifier.name)) => hint
```

Comment:
Isn't this just `shared == identifiers.map(_.name).toSet`
hmm...maybe not much simpler